### ISSUE TITLE
Error while dumping structure.sql file for Pg 9.5.2 

### ISSUE THREAD

**Issue by skatkov** [rails/rails#24652]
### Steps to reproduce

1.  Should store database structure in structure.sql file
2. Should have latest version of postgresql database (9.5.2 in my case)
3. Create a migration with 'rails g migration testMigrateFunctionality'
4. rake db:migrate 

### Expected behavior
Should update structure.sql file.

### Actual behavior
structure.sql file not updated, error shown:

```
pg_dump: invalid option -- i
Try "pg_dump --help" for more information.
rake aborted!
Error dumping database
/Users/sk/.rvm/gems/ruby-2.1.7/gems/activerecord-3.2.22.2/lib/active_record/railties/databases.rake:429:in `block (3 levels) in <top (required)>'
/Users/sk/.rvm/gems/ruby-2.1.7/gems/activerecord-3.2.22.2/lib/active_record/railties/databases.rake:202:in `block (2 levels) in <top (required)>'
/Users/sk/.rvm/gems/ruby-2.1.7/gems/activerecord-3.2.22.2/lib/active_record/railties/databases.rake:196:in `block (2 levels) in <top (required)>'
/Users/sk/.rvm/gems/ruby-2.1.7/bin/ruby_executable_hooks:15:in `eval'
/Users/sk/.rvm/gems/ruby-2.1.7/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => db:structure:dump
(See full trace by running task with --trace)

```
### System configuration
**Rails version**:
Rails 3.2.22.2

**Ruby version**:
ruby 2.1.7p400 (2015-08-18 revision 51632) [x86_64-darwin14.0]

**Comment by vipulnsward** [comment]
Can you confirm if you see this on newer version of Rails 4.2+

**Comment by prathamesh-sonpatki** [comment]
`-i` option is removed from Postgres 9.5. This issue is fixed on master, 4.2.x and 4.1.x in https://github.com/rails/rails/pull/21931.

As Rails 3 is no longer supported it is not backported to 3.2.x. https://github.com/rails/rails/pull/21931#commitcomment-16548022
